docs/docs/performance-tracing.md

```diff
@@ -52,3 +56,26 @@ There are many open tracing compatible backends available. Below is an example o
     ```
 
 4.  Once the build is complete, view your tracing information at [http://localhost:9411](http://localhost:9411)
+
+## Adding your own tracing
+
+The default tracing that comes with Gatsby can give you a good idea of which plugins or stages of the build are slowing down your site. But sometimes, you'll want to trace the internals of your site. Or if you're a plugin author, you might want to trace long operations. 
+
+To provide custom tracing, you can use the `tracing` object, which is present in the args passed to API implementers. This tracing object contains a function called `startSpan`. This simply wraps [open tracing startSpan](https://github.com/opentracing/opentracing-javascript/blob/master/src/tracer.ts#L79), but provides the default `childOf: parentSpan` span args. `startSpan` returns a span object that you must explicitly end by calling its `.finish()` method. For example:
+
+```javascript
+exports.sourceNodes = async ({ actions, tracing, }) => {
+  const span = tracing.startSpan(`foo`)
```

Comment:
Can these spans create spans? This will be a pretty common thing to do e.g. start a span for each API call a source plugin makes.

Also plugins will want to pass a parent span into other libraries which could take advantage of tracing as well e.g. createRemoteFileNode in gatsby-source-filesystem which is used by 4-5 source plugins (at least) https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-filesystem/README.md#createremotefilenode

Reply:
Could potentially add the whole tracer:

```js
{
  tracing: {
    tracer, // <== can be used to manually implement `tracer.startSpan()`
    startSpan,
    parentSpan
  }
}
```

Reply:
Good points. Spans can't create spans, but you can always get the tracer from the span with `const tracer = span.tracer()`. But that feels kinda hacky to me. The two other options are:

1. Require the tracer with `const tracer = require('opentracing').globalTracer()` as we currently do in the Gatsby internals. 
2. Pass the tracer in via the `tracing` object. 

Then you can create a new child span with: `const childSpan = tracer.startSpan({ childOf: parentSpan })`

We should probably do 2) regardless since we don't lose anything. As for what we document, I could go either way. But I guess I'm leaning towards 2) at the moment.

Reply:
@KyleAMathews Great question about passing the parent span to plugin functions. This one is a bit harder. As far as I'm aware, functions exported by plugin authors have no defined interface. It's up to the plugin which argument might accept the parentSpan. Is that right?

Reply:
Yep, might be worth thinking about some kind of standard function signature for random plugin functions in the future. But until then, ad hoc it is.